backend/remote-state/s3/backend.go

```diff
@@ -25,6 +27,14 @@ func New() backend.Backend {
 				Type:        schema.TypeString,
 				Required:    true,
 				Description: "The path to the state file inside the bucket",
+				ValidateFunc: func(v interface{}, s string) ([]string, []error) {
+					// s3 will strip leading slashes from an object, so while this will
+					// technically be accepted by s3, it will break our workspace hierarchy.
+					if strings.HasPrefix(v.(string), "/") {
+						return nil, []error{fmt.Errorf("key cannot contain a leading '/'")}
```

Comment:
This is totally a nit, but "contain a leading slash" reads funny to me since I guess I don't think of a prefix as being "contained".

Suggest instead: `"key must not start with '/'"`